src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -63,6 +63,31 @@ public partial class HubConnection
         private ConnectionState _connectionState;
         private int _serverProtocolMinorVersion;
 
+        /// <summary>
+        /// Occurs when the connection is closed, either by a call to <see cref="StopAsync(CancellationToken)"/> or a connection error.
+        /// </summary>
+        /// <remarks>
+        /// If this event was triggered from a connection error, the <see cref="Exception"/> that occurred will be passed in as the
+        /// sole argument to this handler. If this event was triggered from a call to <see cref="StopAsync(CancellationToken)"/>, then
```

Comment:
You might want to add the server closing the connection gracefully case here as well